src/sardana/macroserver/scan/gscan.py

```diff
@@ -2295,6 +2301,9 @@ def _go_through_waypoints(self):
                     "Moving to waypoint position: %s" % repr(final_pos))
                 motion.move(final_pos)
             finally:
+                # To ensure it is executed.
+                for hook in waypoint.get('post-acq-hooks', []):
```

Comment:
Since at this moment the measurement group may be still acquiring (or extracting the data) I would put the hook execution after the `waitFinish`.

There we have two options:
* **Option 1**: execute them always i.e. even if there was a timeout
* **Option 2**: execute them only if there was no timeout

If we decide for the Option 1 I suggest to do it after the measurement group abort.

What are your opinions about this?

Reply:
If there are no other opinions I will move it there and integrate the PR.

Reply:
I agree

Reply:
Now I see that there is one more point. In case the move gave an error, do we also want to execute the `post-acq` hook? Now it is like this.

Reply:
Good point, should not do it.